src/daft-connect/src/spark_analyzer/expr_resolver.rs

```diff
@@ -0,0 +1,234 @@
+use std::{collections::HashMap, sync::Arc};
+
+use daft_core::datatypes::IntervalValue;
+use daft_dsl::{PlanRef, UnresolvedColumn};
+use daft_logical_plan::LogicalPlanBuilder;
+use spark_connect::{
+    expression::{
+        self as spark_expr,
+        cast::{CastToType, EvalMode},
+        literal::LiteralType,
+        sort_order::{NullOrdering, SortDirection},
+        Literal, UnresolvedFunction,
+    },
+    Expression,
+};
+use tracing::debug;
+
+use super::to_daft_datatype;
+use crate::{
+    error::{ConnectError, ConnectResult},
+    functions::CONNECT_FUNCTIONS,
+    invalid_argument_err, invalid_relation_err, not_yet_implemented,
+};
+
+pub(crate) struct ExprResolver<'a> {
+    pub(crate) plan_nodes: &'a HashMap<i64, LogicalPlanBuilder>,
+    pub(crate) in_join: bool,
+}
+
+impl ExprResolver<'_> {
+    pub(crate) fn resolve_expr(&self, expression: &Expression) -> ConnectResult<daft_dsl::ExprRef> {
+        if let Some(common) = &expression.common {
+            if common.origin.is_some() {
+                debug!("Ignoring common metadata for relation: {common:?}; not yet implemented");
+            }
+        };
+
+        let Some(expr) = &expression.expr_type else {
+            not_yet_implemented!("Expression is required");
+        };
+
+        match expr {
+            spark_expr::ExprType::Literal(l) => to_daft_literal(l),
+            spark_expr::ExprType::UnresolvedAttribute(attr) => {
+                let spark_expr::UnresolvedAttribute {
+                    unparsed_identifier,
+                    plan_id,
+                    is_metadata_column,
+                } = attr;
+                if let Some(is_metadata_column) = is_metadata_column {
+                    debug!("Ignoring is_metadata_column {is_metadata_column} for attribute expressions; not yet implemented");
+                }
+                if self.in_join
+                    && let Some(id) = plan_id
+                {
+                    let plan = self.plan_nodes.get(id);
+
+                    let plan_schema = plan.map(|p| p.schema());
+
+                    let plan_ref = if plan.is_some() {
+                        let id = format!("{id}");
+                        PlanRef::Alias(Arc::from(id))
+                    } else {
+                        PlanRef::Unqualified
```

Comment:
@kevinzwang, i'm pretty sure this is not the intended usage for `PlanRef::Alias`, but it does work the way I want it to. If it's in a join, I add the aliases so i can look up that plan later on inside `fn join()` , then remove them inside [process_join_on](https://github.com/Eventual-Inc/Daft/pull/3849/files#diff-d095d99124e99987567bc211021d2e3f5e101df43e8c8d8698bd42c97a997ef9R745)

Reply:
Hey @universalmind303, would you be able to implement [what we discussed](https://github.com/Eventual-Inc/Daft/pull/3804#issuecomment-2667163089) in the unresolved col PR instead? 

Essentially you would need to
1. add a plan_id: `Option<usize>` value per logical plan node
2. add the `PlanRef::Id(usize)` variant
3. resolve columns by plan ID in `ExprResolver`

Once we have those, it should be very straightforward to implement joins in Spark connect, since we can directly map Spark plan IDs to ours.